Tests/SemanticAnalyzerTests/SemanticAnalyzerTests.swift

```diff
@@ -1,15 +1,223 @@
 import XCTest
 @testable import SemanticAnalyzer
+import Cuckoo
+@testable import AST
+import Diagnostic
+import Source
+import Lexer
 
 final class SemanticAnalyzerTests: XCTestCase {
-  func testExample() {
-    // This is an example of a functional test case.
-    // Use XCTAssert and related functions to verify your tests produce the correct
-    // results.
-    XCTAssert(true)
+
+  // MARK: Fixture -
+  private struct Fixture {
+    let pass = SemanticAnalyzer()
+  }
+
+  // MARK: Mocking Utilities
+  private func buildPassContext(stubEnvironment: (MockEnvironment.Stubbing) -> Void) -> ASTPassContext {
+    var context = ASTPassContext()
+    let environment = MockEnvironment()
+    stub(environment, block: stubEnvironment)
+    context.environment = environment
+    return context
+  }
+
+  // MARK: Dummies
+  var dummyFunctionDeclaration: FunctionDeclaration {
```

Comment:
Should be a function. This way all the usages of `dummyFunctionDeclaration` appear to refer to the same object. Same for the other computed properties below.